src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/AppendHandler.scala

```diff
@@ -19,38 +22,64 @@ class AppendHandler(session: IMAPSession, tag: Tag) extends BaseActor {
 
   override def receive: Receive = {
     case Command.Append(userId, mailbox, flags, date, content) =>
-      if (! date.isEmpty) throw new NotImplementedError("Date parameter for APPEND is still not implemented")
-      val listener = new AppendListener(userId, content)
+      if (!date.isEmpty) throw new NotImplementedError("Date parameter for APPEND is still not implemented")
+      val listener = new AppendListener(self, sender, logger,userId, content)
       val flagsAsString = flags.map(_.mkString("(", " ", ")")).getOrElse("()")
       val length = content.length + content.lines.length - 1
-      session.executeAppendCommand(tag.string, mailbox, flagsAsString, length.toString, listener)
+      logger.debug(s"APPEND receive from sender ${sender.path} on ${self.path}")
       context.become(waitCallback(sender()))
+      Try(session.executeAppendCommand(tag.string, mailbox, flagsAsString, length.toString, listener)) match {
+        case Success(futureResult) =>
+          futureResult.addListener(new IMAPChannelFutureListener {
+            override def operationComplete(future: IMAPChannelFuture): Unit = {
+              logger.debug(s"AppendHandler command completed, success : ${future.isSuccess}")
+              if (!future.isSuccess) {
+                logger.error("AppendHandler command failed", future.cause())
+              }
+
+            }
+          })
+        case Failure(e) =>
+          logger.error("ERROR when executing APPEND COMMAND", e)
+          throw e
+      }
   }
 
   def waitCallback(sender: ActorRef): Receive = {
     case msg@Response.Appended(response) =>
+      logger.debug(s"APPEND reply to sender ${sender.path}")
       sender ! msg
       context.stop(self)
   }
 
 
-  class AppendListener(userId: UserId, content: String) extends IMAPCommandListener {
 
-    import collection.JavaConverters._
+}
+private[command] class AppendListener(self: ActorRef, sender: ActorRef, logger : Logger, userId: UserId, content: String) extends IMAPCommandListener {
```

Comment:
why do you need self? It looks like we used it before without having it as a parameter

Reply:
It's because the AppendListener is now not an inner class of the actor. 
I have done this to avoid the confusion where the sender in the appendListener was DeadLetter.
But if you think it's simpler, i can put it back as an inner class ?

Reply:
no, that's ok, I just missed the actual reason of the change.